### PR TITLE
Taskbundles

### PR DESCRIPTION
**Pull request by gardnervickers**

### CODE REVIEW

**Comment by gardnervickers** [comment]
Should be good to merge

File: src/onyx/redis/tasks.clj [hunk 1]
```diff
@@ -0,0 +1,82 @@
+(ns onyx.redis.tasks
+  (:require [schema.core :as s]
+            [onyx.schema :as os]))
+
+(def UserTaskMapKey
+  (os/build-allowed-key-ns :redis))
+
+(def DatomicReadLogTaskMap
```
Comment (lbradstreet) [inline src/onyx/redis/tasks.clj]: Delete me :)

File: src/onyx/redis/tasks.clj [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+    :schema {:task-map os/TaskMap
+             :lifecycles [os/Lifecycle]}})
+  ([task-name :- s/Keyword
+    kw-fn :- s/Keyword
+    uri :- s/Str
+    task-opts :- {s/Any s/Any}]
+   (connected-task task-name (merge {:onyx/fn kw-fn
+                                           :redis/param? true
+                                           :redis/uri uri}
+                                          task-opts))))
```
Comment (lbradstreet) [inline src/onyx/redis/tasks.clj]: Formatting

**Comment by lbradstreet** [comment]
Left a couple minor comments in the change set.

The other main one that I think we need to address is that I used a different task bundle format for datomic to kafka and redis:
onyx-redis: onyx.redis.tasks
onyx-kafka: onyx.kafka.tasks
onyx-datomic: onyx.plugin.tasks.datomic

The pro for my format is that the tasks are for a plugin, therefore they should be under onyx.plugin along with the plugin code itself.

The pro for your format, is that the tasks don't actually use any plugin code. A third alternative that I kinda like is onyx.tasks.redis.

@MichaelDrogalis what do you think?

**Comment by MichaelDrogalis** [comment]
Personally I like `onyx.tasks.<plugin-name>`, followed by `onyx.plugin.tasks.<plugin-name>`.

**Comment by MichaelDrogalis** [comment]
Remove `.lein-failures` file, add to `.gitignore`.

File: src/onyx/plugin/redis.clj [hunk 3]
```diff
@@ -84,7 +80,7 @@
 (defn take-from-redis
   "conn: Carmine map for connecting to redis
   key: The name of a redis key for looking up the relevant values
-  batch-size: The maximum size of a returned batch
+  batch-size: The maximum size of a returned batchs
```
Comment (MichaelDrogalis) [inline src/onyx/plugin/redis.clj]: Typo

**Comment by MichaelDrogalis** [comment]
Two small comments - gtm.

File: .lein-failures [hunk 4]
```diff
@@ -0,0 +1 @@
+{}
```
Comment (MichaelDrogalis) [inline .lein-failures]: Drop file.

**Pull request merged**